modules/gdscript/gdscript_parser.cpp

```diff
@@ -1721,6 +1721,20 @@ GDScriptParser::Node *GDScriptParser::_reduce_expression(Node *p_node, bool p_to
 				return op;
 
 			} else if (op->op == OperatorNode::OP_CALL) {
+
+				// range function step argument must not be 0
+				if (op->arguments[0]->type == Node::TYPE_BUILT_IN_FUNCTION) {
+					if (static_cast<BuiltInFunctionNode *>(op->arguments[0])->function == GDScriptFunctions::GEN_RANGE) {
+						if (op->arguments.size() == 4 && op->arguments[3]->type == Node::Type::TYPE_CONSTANT) {
+							ConstantNode *c = static_cast<ConstantNode *>(op->arguments[3]);
+							if (c->value.operator double() == 0) {
+								_set_error("Step argument must not be 0.");
```

Comment:
Here's a clearer message which can be useful for long lines:

```suggestion
								_set_error("The range() step argument must not be equal to 0.");
```